### PR TITLE
Enable strong-migrations in all environments

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

When we merged #5522 we weren't familiar with the gem and opted to enable it only in development, but to be effective this gem should be enabled in production as well, being able to catch unsafe migrations if they somehow skip our PR review process.

By merging #6242 we broke deploy because the https://github.com/ankane/strong_migrations gem wasn't enabled in production. @mstruve quickly fixed it with #6277

This should fix the underline problem we have triggered

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -92,6 +92,7 @@ gem "sprockets", "~> 3.7" # Sprockets is a Rack-based asset packaging system
 gem "staccato", "~> 0.5" # Ruby Google Analytics Measurement
 gem "storext", "~> 3.1" # Add type-casting and other features on top of ActiveRecord::Store.store_accessor
 gem "stripe", "~> 5.15" # Ruby library for the Stripe API
+gem "strong_migrations", "~> 0.6" # Catch unsafe migrations in development
```
Comment (mstruve) [inline Gemfile]: we should remove the development part from the comment but I'm also fine with this and we can clean up later

Comment (nickytonline) [reply Gemfile]: So the impact of moving the `strong_migrations` gem in the Gemfile is so it's applied in any environment?

Comment (mstruve) [reply Gemfile]: yes

File: config/initializers/strong_migrations.rb [hunk 2]
```diff
@@ -1,7 +1,5 @@
-if Rails.env.test? || Rails.env.development?
```
Comment (mstruve) [inline config/initializers/strong_migrations.rb]: Good catch!

Comment (rhymes) [reply config/initializers/strong_migrations.rb]: thanks to @citizen428 :D!

**Pull request merged**